pilot/pkg/bootstrap/server.go

```diff
@@ -389,42 +399,15 @@ func (s *Server) httpServerReadyHandler(w http.ResponseWriter, _ *http.Request)
 	w.WriteHeader(http.StatusOK)
 }
 
-func (s *Server) initDiscoveryService(args *PilotArgs) error {
-	s.mux.HandleFunc("/ready", s.httpServerReadyHandler)
-
-	s.EnvoyXdsServer.InitDebug(s.mux, s.ServiceController(), args.DiscoveryOptions.EnableProfiling, s.injectionWebhook)
-
-	// When the mesh config or networks change, do a full push.
-	s.environment.AddMeshHandler(func() {
-		// Inform ConfigGenerator about the mesh config change so that it can rebuild any cached config, before triggering full push.
-		s.EnvoyXdsServer.ConfigGenerator.MeshConfigChanged(s.environment.Mesh())
-		s.EnvoyXdsServer.ConfigUpdate(&model.PushRequest{
-			Full:   true,
-			Reason: []model.TriggerReason{model.GlobalUpdate},
-		})
-	})
-	s.environment.AddNetworksHandler(func() {
-		s.EnvoyXdsServer.ConfigUpdate(&model.PushRequest{
-			Full:   true,
-			Reason: []model.TriggerReason{model.GlobalUpdate},
-		})
-	})
-
-	if err := s.initEventHandlers(); err != nil {
-		return err
-	}
-
-	// Implement EnvoyXdsServer grace shutdown
-	s.addStartFunc(func(stop <-chan struct{}) error {
-		s.EnvoyXdsServer.Start(stop)
-		return nil
-	})
-
+// initDebugServer initializes debug end points.
+func (s *Server) initDebugServer(args *PilotArgs, wh *inject.Webhook) error {
+	s.debugMux.HandleFunc("/ready", s.httpServerReadyHandler)
```

Comment:
Technically '/ready' is not part of the debug server - but readiness. 

I remember in a distant past the 'debug' interfaces had a flag that allowed disabling them - maybe we should bring it back. 

I think the /ready may share the same port with monitoring - both need to be enabled.

Reply:
Changed it to IstiodAdminServer combining monitoring, debug and readiness as is being done today